optuna/samplers/_tpe/sampler.py

```diff
@@ -389,17 +389,20 @@ def _sample_from_gmm(
         active = np.argmax(self._rng.multinomial(1, weights, size=size), axis=-1)
         trunc_low = (low - mus[active]) / sigmas[active]
         trunc_high = (high - mus[active]) / sigmas[active]
-        while True:
-            samples = truncnorm.rvs(
-                trunc_low,
-                trunc_high,
-                size=size,
-                loc=mus[active],
-                scale=sigmas[active],
-                random_state=self._rng,
+        samples = np.full(size, fill_value=high + 1.0, dtype=np.float64)
```

Comment:
Sorry for the late comment but I now recalled that [`numpy.where`](https://numpy.org/doc/stable/reference/generated/numpy.where.html) broadcasts. You don't have to allocate memory for `size` even. This should address the degradation for the general case w/o resampling.
```suggestion
        samples = np.full((), fill_value=high + 1.0, dtype=np.float64)
```

Reply:
Actually, I observed that the suggested code https://github.com/optuna/optuna/pull/1562#discussion_r462118803 imposes an only negligible overhead to the current master, and I think we don't need to care the degradation too much.
Nevertheless, the newer code seems clearly better and let me adopt the suggestion.